### ISSUE TITLE
`object_usage_linter()` should set `useFancyQuotes=FALSE`

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1498]
Compare:

```r
str <- "foo <- function() {\n  x\n}\n"

options(useFancyQuotes = FALSE)
lintr::lint(text = str, linters = lintr::object_usage_linter())
# <text>:2:3: warning: [object_usage_linter] no visible binding for global variable 'x'.
#   x
#   ^
options(useFancyQuotes = TRUE)
lintr::lint(text = str, linters = lintr::object_usage_linter())
# <text>:2:3: warning: [object_usage_linter] no visible binding for global variable ‘x’.
#   x
#   ^
```

I think we should follow suit of `testthat` and turn off quotes here -- makes the lint appear differently depending on global options & it's harder to manipulate the message with non-ASCII characters.

The fix is pretty trivial to implement; filing here first as a gut check that this is the right approach.

**Comment by MichaelChirico** [comment]
That's basically right, but it's still something in our control that allows us to be more consistent. I don't think we use fancy quotes in any messages generated by us. And the fix is one line (wrap `codetools` call with `with_options(list(useFancyQuotes = FALSE), ...)`)
